### PR TITLE
Clarify firing of `import_notification_stream` in doc comment

### PR DESCRIPTION
**Pull request by ffarall**
# Description

Updates the doc comment on the `import_notification_stream` to make its behaviour clearer.

Closes [Unexpected behaviour of block `import_notification_stream`](https://github.com/paritytech/polkadot-sdk/issues/5596).

## Integration

Doesn't apply.

## Review Notes

The old comment docs caused some confusion to myself and some members of my team, on when this notification stream is triggered. This is reflected in the linked [issue](https://github.com/paritytech/polkadot-sdk/issues/5596), and as discussed there, this PR aims to prevent this confusion in future devs looking to make use of this functionality.

# Checklist

* [x] My PR includes a detailed description as outlined in the "Description" and its two subsections above.
* [ ] My PR follows the [labeling requirements](
https://github.com/paritytech/polkadot-sdk/blob/master/docs/contributor/CONTRIBUTING.md#Process
) of this project (at minimum one label for `T` required)
    * External contributors: ask maintainers to put the right label on your PR.
* [x] I have made corresponding changes to the documentation (if applicable)
* [x] I have added tests that prove my fix is effective or that my feature works (if applicable)

You can remove the "Checklist" section once all have been checked. Thank you for your contribution!

### CODE REVIEW

**Review by bkchr — APPROVED** [review]

**Review by lexnv — APPROVED** [review]

File: substrate/client/api/src/client.rs [hunk 1]
```diff
@@ -65,9 +65,15 @@ pub trait BlockOf {
 pub trait BlockchainEvents<Block: BlockT> {
 	/// Get block import event stream.
 	///
-	/// Not guaranteed to be fired for every imported block, only fired when the node
-	/// has synced to the tip or there is a re-org. Use `every_import_notification_stream()`
+	/// Not guaranteed to be fired for every imported block. Use `every_import_notification_stream()`
 	/// if you want a notification of every imported block regardless.
+	///
+	/// This notification stream is fired...
```
Comment (michalkucharczyk) [inline substrate/client/api/src/client.rs:71]:
```suggestion
	/// This notification stream is fired:
```

**Review by michalkucharczyk — APPROVED** [review]

**Comment by ffarall** [comment]
@bkchr prdoc added. I wasn't sure before if it was needed, given that it's a change in the comments only. But it's there now.

File: prdoc/pr_5811.prdoc [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+title: ...
+
+doc:
+  - audience: Node Dev
+    description: |
+      "Updates the doc comment on the `import_notification_stream` to make its behaviour clearer. Now it specifically states that this notification stream is fired on every import notification after the initial sync, and only when there are re-orgs in the initial sync."
+
+crates:
+  - name: sc-client-api
+    bump: minor
```
Comment (bkchr) [inline prdoc/pr_5811.prdoc:13]:
```suggestion
    bump: patch
```

**Review by bkchr — APPROVED** [review]

File: prdoc/pr_5811.prdoc [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Schema: Polkadot SDK PRDoc Schema (prdoc) v1.0.0
+# See doc at https://raw.githubusercontent.com/paritytech/polkadot-sdk/master/prdoc/schema_user.json
+
+title: ...
```
Comment (bkchr) [inline prdoc/pr_5811.prdoc:4]:
```suggestion
title: "Improve `import_notification_stream` documentation" 
```

**Pull request merged**